PetrolEngine.cpp

```diff
@@ -12,9 +12,11 @@ PetrolEngine::PetrolEngine(int power, float capacity, int gears)
 
 void PetrolEngine::changeGear(int gear)
 {
-    // TODO: Add checking if gear is between -1 and gears_
-    // -1 is for REAR
-    // 0 is for NEUTRAL
-    currentGear_ = gear;
-    std::cout << __FUNCTION__ << std::endl;
+    if(-1 <= gear && gear <= PetrolEngine::gears_)
+        currentGear_ = gear;
+    else if(gear < -1)
+        currentGear_ = -1;
+    else
+        currentGear_ = PetrolEngine::gears_;
```

Comment:
Hmm... raczej bym tutaj widział jakiś błąd, bo taka sytuacja nie powinna nigdy wystąpić. Rzucenie wyjątku będzie idealne. Jeśli jeszcze nie wiecie jak to chociaż wypiszcie tutaj na ekran że jest błąd. Dla warunku gear < -1 również.

Reply:
poprawione ;)